astropy/units/format/ogip.py

```diff
@@ -206,10 +213,6 @@ def p_unit_expression(p):
                             | UNIT OPEN_PAREN complete_expression CLOSE_PAREN power numeric_power
                             | OPEN_PAREN complete_expression CLOSE_PAREN power numeric_power
             """
-            # If we run p[1] in cls._functions, it will try and parse each
```

Comment:
Why delete this comment? I think it is useful to know why `p1_str` is done.

Reply:
Previously `cls._functions` was a `list`, so if `p[1]` was not in it then `p[1] in cls._functions` would have looped over all the elements and tried to perform expensive unit conversion at every step. But now `cls._functions` is a `frozenset`, so `p[1] in cls._functions` is fast whether or not `p[1]` is in there.

Reply:
Previously I had changed the type of `OGIP._functions` from `list` to `frozenset`, but I've undone that change and the related comment deletion. I am convinced that `OGIP._functions` should be a `frozenset`, but `p_unit_expression()` should then be updated accordingly. The code changes there should be simple, but it looks like the test coverage is incomplete. It is better to first increase coverage so that code changes could be made with confidence, but updating the tests might take more time, so that's best done in a separate follow-up.